### ISSUE TITLE
Wetland pattern is shown too soon

### ISSUE THREAD

**Issue by jeisenbe** [gravitystorm/openstreetmap-carto#4071]
### Expected behavior
- The wetland pattern should be shown after the fill color. 
- The wetland pattern should not be shown before rivers and large lakes
- Patterns add noise and are hard to interpret when areas are very small, since the wetland pattern is only intelligible when the area is more than about 16 x 16 pixels. 
- In contrast, even 1 pixel of fill color can show that something is mapped, and our vegetation/natural area colors are designed to mix well even at sub-pixel scale.
- This suggest that patterns should be shown 4 or 5 zoom levels later than the fill color.

### Actual behavior
- The wetland pattern is shown from z5, which is the same zoom level as most of the `wetland=*` fill colors
- Rivers are not shown until z8, so from z5 to z7 small wetlands can be shown but not moderately large rivers
- Lakes are only shown when >32 pixels in size (e.g. 8x4), while even 1 pixel wetlands are attempted to be rendered (though the pattern won't be visible unless there are many >1 pixel size areas next to each other)
- However, wetland=tidalflat and natural=mud fill color is only shown from z9. This means that from z5 to z8 these features have the wetland pattern but no fill color, most noticable when they are mapped on land.

### Recommended solution
- Change the wetland pattern back to z10, as it was previously. At this zoom level many wetlands are big enough to be intelligible, and rivers are wide enough to still be visible in spite of the busy wetland pattern (though it would help if rivers were darker)
- Consider showing wetland=tidalflat / natural=mud fill color from z8 or perhaps the same as other fill colors (though there are serious issues with the z5 to z7 landcover layer which prevent us from getting a good rendering without further changes)

### Links and screenshots illustrating the problem
z6 - https://www.openstreetmap.org/#map=6/-4.741/139.164
- Wetland pattern is barely visible, even though these mangroves are 40 x 200km in extent
![export-27](https://user-images.githubusercontent.com/42757252/76478581-816e4100-644c-11ea-88ed-0704eba696dc.png)

z8 - rivers and tidal channels paths are obscured by the strong pattern
![export-26](https://user-images.githubusercontent.com/42757252/76478592-87fcb880-644c-11ea-91c3-c4afe68c0036.png)

z10 - some of the small rivers and tidal channels are still hard to follow, but the large ones are at least clear at this zoom level.
![export-21](https://user-images.githubusercontent.com/42757252/76478587-859a5e80-644c-11ea-8a7f-c8f26c590994.png)

**Comment by jeisenbe** [comment]
See also the tiny blue dashes created by the small* wetland areas in Latvia. These sometimes appear and then disappear at the next zoom level, since the pattern has wide gaps:

*(Actually, the wetland areas are pretty large since they are at least a few pixels at this zoom level, but too small for the pattern to be clearly seen)

z5 current
Current (32 pixel area limit for water, 0.01 pixel are limit for landcover, e.g. woodland)
<img width="173" alt="z5-latvia-master-water-filtering" src="https://user-images.githubusercontent.com/42757252/76277513-4cd37b80-62cc-11ea-9fce-340cc3347baf.png">

z6 current
![z6-latvia-master-filtered-32pixels](https://user-images.githubusercontent.com/42757252/76278265-8ad19f00-62ce-11ea-8a66-1977b7e493b9.png)

z7 current
<img width="621" alt="z7-latvia-master-water-filtering" src="https://user-images.githubusercontent.com/42757252/76278310-b3599900-62ce-11ea-95e2-c675df613ea8.png">

z7 current 3x export SE Latvia (lots of wetlands):
Current - 3x export, SE Latvia
![z7-latvia-export-3x-master-water-filtering](https://user-images.githubusercontent.com/42757252/76277495-3fb68c80-62cc-11ea-942e-5df7f64bb650.png)

As mentioned in #4067, it is strange to show a 1 pixel area of wetland, but only show >32 pixel areas of natural=water.

**Comment by kocio-pl** [comment]
I think moving them later makes sense, but "4 or 5 zoom levels later than the fill color" sounds to me like way too much.

**Comment by jeisenbe** [comment]
z10 is 5 zoom levels later than z5, which is where the fill color is
shown. That is also the zoom level where we used to render wetland
patterns. It would still be 3 zoom levels sooner than any other
pattern.

I'm willing to consider z9 for wetlands - that's still one zoom level
later than rivers are shown. But z8 does not work well for areas near
the equator, especially since small rivers are just appearing then.

To show wetlands at z8 we would need to use a very different pattern
which was much finer at z8/z9: perhaps much shorter and thinner lines,
closer together. That might be possible with 2x "retina" tiles but not
with the current standard-resolution rendering.

**Comment by imagico** [comment]
z10 is IMO a good starting zoom levels for the wetland pattern.  I considered that for the ac-style and came to the same conclusion.

Note the largest continuous wetland areas are tidalflats and these are rendered with a color fill from z9 - originally this was specifically meant to have these being shown one zoom level earlier in a non-obtrusive form.  Now the wetland pattern starts at z5 and the fill is added at z9 which looks just weird.

Redesigning the wetland pattern would be something worth considering - the current design is mostly inspired by the German tradition but there are other styles that might be worth exploring.  But in any case it is unlikely that a pattern would be suitable for display at z<10.
